packages/core/src/odata/common/request-builder/batch/batch-request-serializer.ts

```diff
@@ -0,0 +1,98 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+
+import voca from 'voca';
+import { MethodRequestBuilderBase } from '../request-builder-base';
+import { BatchChangeSet } from './batch-change-set';
+import { BatchRequestBuilder } from './batch-request-builder';
+
+/**
+ * Serialize change set to string.
+ * @param changeSet - Change set containing a collection of write operations.
+ * @param changeSetBoundary Boundary separating parts in the serialized request.
+ * @returns The serialized string representation of a change set.
+ */
+export function serializeChangeSet(
+  changeSet: BatchChangeSet
+): string | undefined {
+  if (changeSet.requests.length) {
+    return [
+      `Content-Type: multipart/mixed; boundary=${changeSet.boundary}`,
+      '',
+      `--${changeSet.boundary}`,
+      changeSet.requests
+        .map(request => serializeRequest(request))
+        .join(`\n--${changeSet.boundary}\n`),
+      `--${changeSet.boundary}--`
+    ].join('\n');
+  }
+}
+
+/**
+ * Serialize a multipart request to string.
+ * @param request One of [[GetAllRequestBuilder | getAll]], [[GetByKeyRequestBuilder | getByKey]], [[CreateRequestBuilder | create]], [[UpdateRequestBuilder | update]] or [[DeleteRequestBuilder | delete]] request builder.
+ * @returns The serialized string representation of a multipart request, including the multipart headers.
+ */
+export function serializeRequest(request: MethodRequestBuilderBase) {
+  const odataRequest = request.build();
+  const headers = {
+    ...odataRequest.defaultHeaders(),
+    ...odataRequest.eTagHeaders(),
+    ...odataRequest.customHeaders()
+  };
+  const requestHeaders = Object.entries(headers).map(
+    ([key, value]) => `${voca.titleCase(key)}: ${value}`
+  );
+  return [
+    'Content-Type: application/http',
+    'Content-Transfer-Encoding: binary',
+    '',
+    `${request.requestConfig.method.toUpperCase()} /${request.relativeUrl()} HTTP/1.1`,
+    ...(requestHeaders.length ? requestHeaders : ['']),
+    '',
+    ...getPayload(request),
+    ''
+  ].join('\n');
+}
+
+function getPayload(request: MethodRequestBuilderBase): string[] {
+  return request.requestConfig.method !== 'get'
+    ? [JSON.stringify(request.requestConfig.payload)]
+    : [];
+}
+
+/**
+ * Serialize a batch request to string. This is used for the batch request payload when executing the request.
+ * @param request - Batch request to serialize.
+ * @returns String representation of the batch request.
+ */
+export function serializeBatchRequest(request: BatchRequestBuilder): string {
+  const serializedSubRequests = request.requests
+    .map(subRequest =>
+      subRequest instanceof MethodRequestBuilderBase
+        ? serializeRequest(subRequest)
+        : serializeChangeSet(subRequest)
+    )
+    .filter(validRequest => !!validRequest)
+    .join(`\n--${request.requestConfig.boundary}\n`);
+
+  return serializedSubRequests
```

Comment:
under which condition can the "or" part happen i.e. return serializedSubRequests? If the request.request is empty it will return an empty list, which is falsy I guess, but then the return type string is not correct. Empty string would be more correct, so perhaps also add a `join('\n')`?

Reply:
At this point `serializedSubRequests` is a string. It can be an empty string in which case it would be falsy. The join is in line 76.